codalab/server/bundle_manager.py

```diff
@@ -213,9 +213,15 @@ def _make_bundle(self, bundle):
             )
             for dep in bundle.dependencies:
                 parent_bundle_link_url = parent_bundle_link_urls.get(dep.parent_uuid)
-                parent_bundle_path = parent_bundle_link_url or os.path.normpath(
-                    self._bundle_store.get_bundle_location(dep.parent_uuid)
-                )
+                try:
+                    parent_bundle_path = parent_bundle_link_url or os.path.normpath(
+                        self._bundle_store.get_bundle_location(dep.parent_uuid)
+                    )
+                except NotFoundError:
+                    raise Exception(
```

Comment:
Is there a more specific type of Exception to raise?  Basically, is this treated as a user error or a system error?

Reply:
This is a system error, but I'm not sure what kind of exception to raise here. I'm just following the convention used elsewhere in this file of raising `Exception` classes on error -- no other types of exceptions are raised here. We can probably refactor them into system-error-specific Exception subclasses in a future PR.